gpytorch/variational/lmc_variational_strategy.py

```diff
@@ -0,0 +1,149 @@
+
+    Example:
+        >>> class LMCMultitaskGP(gpytorch.models.ApproximateGP):
+        >>>     '''
+        >>>     3 latent functions
+        >>>     5 output dimensions (tasks)
+        >>>     '''
+        >>>     def __init__(self):
+        >>>         # Each latent function shares the same inducing points
+        >>>         # We'll have 32 inducing points, and let's assume the input dimensionality is 2
+        >>>         inducing_points = torch.randn(32, 2)
+        >>>
+        >>>         # The variational parameters have a batch_shape of [3] - for 3 latent functions
+        >>>         variational_distribution = gpytorch.variational.MeanFieldVariationalDistribution(
+        >>>             inducing_points.size(-1), batch_shape=torch.Size([3]),
+        >>>         )
+        >>>         variational_strategy = gpytorch.variational.LMCVariationalStrategy(
+        >>>             gpytorch.variational.VariationalStrategy(
+        >>>                 inducing_points, variational_distribution, learn_inducing_locations=True,
+        >>>             ),
+        >>>             num_tasks=5,
+        >>>             num_latents=3,
+        >>>             latent_dim=0,
+        >>>         )
+        >>>
+        >>>         # Each latent function has its own mean/kernel function
+        >>>         super().__init__(variational_strategy)
+        >>>         self.mean_module = gpytorch.means.ConstantMean(batch_shape=torch.Size([3]))
+        >>>         self.covar_module = gpytorch.kernels.ScaleKernel(
+        >>>             gpytorch.kernels.RBFKernel(batch_shape=torch.Size([3])),
+        >>>             batch_shape=torch.Size([3]),
+        >>>         )
+        >>>
+        >>> # Model output: n x 5
+    """
+
+    def __init__(
+        self, base_variational_strategy, num_tasks, num_latents=1, latent_dim=0,
+    ):
+        Module.__init__(self)
+        self.base_variational_strategy = base_variational_strategy
+        self.num_tasks = num_tasks
+        batch_shape = self.base_variational_strategy._variational_distribution.batch_shape
+
+        # Check if no functions
+        if latent_dim >= 0:
+            latent_dim - len(batch_shape)
+        if not (batch_shape[latent_dim] == num_latents or batch_shape[latent_dim] == 1):
+            raise RuntimeError(
+                f"Mismatch in num_latents: got a variational distribution of batch shape {batch_shape}, "
+                f"expected the function dim {self.latent_dim} to be {self.num_latents}."
+            )
+        self.num_latents = num_latents
+        self.latent_dim = latent_dim
+
+        # Make the batch_shape
+        self.batch_shape = list(batch_shape)
+        del self.batch_shape[self.latent_dim]
+        self.batch_shape = torch.Size(self.batch_shape)
+
+        # LCM coefficients
+        raw_lmc_coefficients = torch.randn(*batch_shape, self.num_tasks).tril_()
```

Comment:
why tril?

Reply:
that was leftover code. i caught it once i started writing an example notebook 😬